arviz/plots/pairplot.py

```diff
@@ -158,13 +163,19 @@ def plot_pair(
                 divergent_data, var_names=("diverging",), combined=True
             )
             diverging_mask = np.squeeze(diverging_mask)
+        elif hasattr(data, "sample_stats_prior") and hasattr(data.sample_stats_prior, "diverging"):
+            divergent_data = convert_to_dataset(data, group="sample_stats_prior")
+            _, diverging_mask = xarray_to_ndarray(
+                divergent_data, var_names=("diverging",), combined=True
+            )
+            diverging_mask = np.squeeze(diverging_mask)
```

Comment:
The elif will not achieve the desired behaviour. The cases to take into account are:

* `group="posterior"` --> divergent_data is taken from `sample_stats`
* `group="prior"` --> divergent_data is taken from `sample_stats_prior`
* group in anything else --> no divergent data is plotted

This can use `sample_stats` when group is prior or sample_stats_prior when group is posterior (if there is no sample stats group)

Reply:
@OriolAbril Yes, I realized my mistake and rectified in the new commit just now. Kindly review that solution.